### PR TITLE
feat: [PL-28359]: hotfix for fetching file from branch having slash

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
with branch having slash, getFile by branch does not work for bb on-prem, that's why firstly latest commit has been found on branch and then file is fetched via commitId.

Tested on local by creating a file on bb server in such a branch, fetching it and updating it.
get file has been tested for github and bb saas as well.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37693)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37693)

**Review by adivishy1 — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37693)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java [hunk 1]
```diff
@@ -215,8 +215,15 @@ public FileContent getFileContent(
     String slug = scmGitProviderHelper.getSlug(scmConnector);
     final GetFileRequest.Builder gitFileRequestBuilder =
         GetFileRequest.newBuilder().setPath(gitFilePathDetails.getFilePath()).setProvider(gitProvider).setSlug(slug);
-    if (isNotEmpty(gitFilePathDetails.getBranch())) {
+    if (isNotEmpty(gitFilePathDetails.getBranch()) && !gitFilePathDetails.getBranch().contains("/")) {
       gitFileRequestBuilder.setBranch(gitFilePathDetails.getBranch());
+    } else if (isNotEmpty(gitFilePathDetails.getBranch())) {
```
Comment (meenakshi-harness) [inline 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java:220]: Same condition is checked twice. We can check it once and move the other condition in nested if-else

**Comment by mohitg0795** [comment]
*[`952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java` line 218 at r1](https://reviewable.io/reviews/harness/harness-core/37693#-NCPALD38KCGaWZ039QK:-NCPALD4DXf6CGTgrHnK:b-xcuao1) ([raw file](https://github.com/harness/harness-core/blob/9d9cedf0c1e47f41d1a831cbc07719dafc12f804/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L218)):*
> ```Java
>     final GetFileRequest.Builder gitFileRequestBuilder =
>         GetFileRequest.newBuilder().setPath(gitFilePathDetails.getFilePath()).setProvider(gitProvider).setSlug(slug);
>     if (isNotEmpty(gitFilePathDetails.getBranch()) && !gitFilePathDetails.getBranch().contains("/")) {
> ```

Condition for branch with slash should be moved to a method. Also, this check is only for bitbucket. For others, it shouldn't be done.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37693)* status: 3 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java` line 218 at r1](https://reviewable.io/reviews/harness/harness-core/37693#-NCPALD38KCGaWZ039QK:-NCPE7M3BtJ9wzemwfAU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9d9cedf0c1e47f41d1a831cbc07719dafc12f804/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L218)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Condition for branch with slash should be moved to a method. Also, this check is only for bitbucket. For others, it shouldn't be done.
</blockquote></details>

Done.
___
*[`952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java` line 220 at r1](https://reviewable.io/reviews/harness/harness-core/37693#-NCP76PPgBNpTeV-DiLh-r1-220:-NCPE7eq-jWp04RpjilJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9d9cedf0c1e47f41d1a831cbc07719dafc12f804/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L220)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Same condition is checked twice. We can check it once and move the other condition in nested if-else
</blockquote></details>

Done.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37693)* status: 1 file(s) to review. 1. unresolved discussions.

**Comment by mohitg0795** [comment]
*[`952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java` line 236 at r3](https://reviewable.io/reviews/harness/harness-core/37693#-NCPGStqCAhP148Noyax:-NCPGStqCAhP148Noyay:buu25ot) ([raw file](https://github.com/harness/harness-core/blob/37d1be0c7fe4383bc3ded505bc95b516636f5a3b/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L236)):*
> ```Java
> 
>   private boolean checkIfBranchIsHavingSlashForBB(ScmConnector scmConnector, String branchName) {
>     return scmConnector.getConnectorType().equals(ConnectorType.BITBUCKET) && branchName.contains("/");
> ```

can we just reverse the equals statement here: BITBUCKET.equals(connectorType). Just better to avoid any chances of NPE.

**Pull request merged**